modules/new_serverpod_auth/serverpod_auth_email_account/serverpod_auth_email_account_server/lib/src/business/email_accounts.dart

```diff
@@ -0,0 +1,657 @@
+
+/// Email account management functions.
+abstract final class EmailAccounts {
+  /// Returns the [AuthUser]'s ID upon successful login.
+  ///
+  /// Throws [EmailAccountLoginException] for expected error cases.
+  static Future<UuidValue> login(
+    final Session session, {
+    required String email,
+    required final String password,
+    final Transaction? transaction,
+  }) async {
+    return session.transactionOrSavepoint((final transaction) async {
+      email = email.trim().toLowerCase();
+
+      var account = await EmailAccount.db.findFirstRow(
+        session,
+        where: (final t) => t.email.equals(email),
+        transaction: transaction,
+      );
+
+      if (await _hasTooManyFailedSignIns(session, email)) {
+        throw EmailAccountLoginException(
+          reason: EmailAccountLoginFailureReason.tooManyAttempts,
+        );
+      }
+
+      try {
+        account ??= await _importExistingUser(
+          session,
+          email: email,
+          password: password,
+          transaction: transaction,
+        );
+      } catch (_) {
+        await _logFailedSignIn(session, email);
+
+        rethrow;
+      }
+
+      if (account == null) {
+        throw EmailAccountLoginException(
+          reason: EmailAccountLoginFailureReason.invalidCredentials,
+        );
+      }
+
+      if (!PasswordHash.validateHash(
+        email: email,
+        password: password,
+        hash: account.passwordHash,
+      )) {
+        await _logFailedSignIn(session, email);
+
+        throw EmailAccountLoginException(
+          reason: EmailAccountLoginFailureReason.invalidCredentials,
+        );
+      }
+
+      return account.authUserId;
+    }, transaction: transaction);
+  }
+
+  /// Returns the result of the operation and a process ID for the account request.
+  ///
+  /// An account request is only created if the `result` is [EmailAccountRequestResult.accountRequestCreated].
+  /// In all other cases `accountRequestId` will be `null`
+  ///
+  /// The caller should ensure that the actual result does not leak to the outside / client.
+  /// Instead clients generally should always see a message like "If this email was not registered already,
+  /// a new account has been created and a verification email has been sent".
+  /// This prevents the endpoint being misused to scan for registered/valid email addresses.
+  ///
+  /// The caller might decide to initiate a password reset (per email, not in the client response), to help users which try
+  /// to register but already have a valid account.
+  ///
+  /// In the success case of [EmailAccountRequestResult.accountRequestCreated], the caller may store additional information
+  /// attached to the `accountRequestId`, which will be returned from [verifyAccountRequest] later on.
+  static Future<
+      ({
+        EmailAccountRequestResult result,
+        UuidValue? accountRequestId,
+      })> requestAccount(
```

Comment:
I would like to rename these and align them with the password reset methods.


Maybe

- startAccountCreation
- verifyAccountCreation (or `checkAccountCreation`?)
- completeAccountCreation

Then for the reset flow

- startPasswordReset
- verifyPasswordReset
- completePasswordReset

Reply:
Updated the naming pattern.